oscarlab/graphene
# Cleanup .gitignore files

## Affected components


## Description of the changes (reasons and measures)
This PR fixes various things related to rules in .gitignore files:
* Adds missed files
* Excludes files present in the repository
* Fixes unintentionally recursive rules
* Extracts rules with editors' cache files to the main .gitignore

I also deleted two rules from `Pal/.gitignore`: `libpal.a.backup` and `libpal.so.backup`. I couldn't find any reference to them in the codebase, I guess those are remains from some older version of the source?

I also skipped updating a few apps from shim/test/apps, because this folder doesn't seem to be actively maintained (and only fixed already-existing gitignores)

## How to test this PR? (if applicable)

Build and run all tests and see whether `git status` reports any new files (it shouldn't).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/493)
